src/agent.cc

```diff
@@ -112,6 +112,20 @@ void Agent::lifetime(int n_timesteps) {
   lifetime_ = n_timesteps;
 }
 
+void Agent::lifetime_force(int n_timesteps) {
+  try{
+    lifetime(n_timesteps);
+  }
+  catch (ValueError e){
+    if(enter_time_+n_timesteps <= context()->time()){
+      lifetime(context()->time() - enter_time_ + 1);
+    }
+    else{
+      lifetime_ = n_timesteps;    
```

Comment:
As the only check in the `lifetime()` method is if the number of timestep (n_timesteps) is negative...
if `lifetime()` throws an error your implementation will actually overwrite the check and still set the lifetime to a negative's value.
If I understand correctly, is this intended ?

Reply:
No - `lifetime()` checks the value of `enter_time` not `n_timesteps`....

Reply:
True !

I think I understand now. facility that have not being deployed have a `enter_time_ = -1` but once deployed it is set to the actual `enter_time`.
so if `lifetime()` throw an error that means the facility has been deployed
if the facility has been deployed, 2cases:
- the new lifetime ends before the actual time (so we set it to the end at the next timestep)
- the new lifetime ends after the actual time, we just update the lifetime...